### PR TITLE
Document auto_import as a feature

### LINKED ISSUE #6225: Document auto-import
In my project we prefer to use `use crate::foo::bar`, even if `use super::bar` would work. Rust Analyzer always suggests `super` imports if it can. It would be great to:

* Have an action to turn `super` into `create` where possible.
* Have an option to insert `crate` references in preference to `super`.

### PR DESCRIPTION
**Pull request by Veykril**
Closes #6225

### CODE REVIEW

File: crates/assists/src/handlers/auto_import.rs [hunk 1]
```diff
@@ -6,6 +6,61 @@ use crate::{
     AssistContext, AssistId, AssistKind, Assists, GroupLabel,
 };
 
+// Feature: Import Insertion
+//
+// Using the `auto-import` assist it is possible to insert missing imports for unresolved Items.
```
Comment (lnicola) [inline crates/assists/src/handlers/auto_import.rs:11]:
```suggestion
// Using the `auto-import` assist it is possible to insert missing imports for unresolved items.
```

File: crates/assists/src/handlers/auto_import.rs [hunk 2]
```diff
@@ -6,6 +6,61 @@ use crate::{
...
+// When inserting an import it will do so in a structured manner by keeping imports grouped,
+// separated by a newline in the following order:
+//
+// - Std and Core
```
Comment (lnicola) [inline crates/assists/src/handlers/auto_import.rs:15]:
```suggestion
// - `std` and `core`
```

File: crates/assists/src/handlers/auto_import.rs [hunk 3]
```diff
@@ -6,6 +6,61 @@ use crate::{
...
+//
+// .Merge Behaviour
+//
+// It is possible to configure how use-trees are merged with the `importMergeBehaviour` setting.
+// It has the following configurations:
+//
+// - `full`: This setting will cause auto-import to always completely merge use-trees that share the
+//  same path prefix while also merging inner trees that share the same path-prefix. This kind of
+//  nesting is only supported in Rust versions later than 1.24.
+// - `last`: This setting will cause auto-import to merge use-trees as long the resulting tree will
```
Comment (lnicola) [inline crates/assists/src/handlers/auto_import.rs:43]:
```suggestion
// - `last`: This setting will cause auto-import to merge use-trees as long as the resulting tree will
```

File: crates/assists/src/handlers/auto_import.rs [hunk 4]
```diff
@@ -6,6 +6,61 @@ use crate::{
...
+//  only contain a nesting of single segment paths at the very end.
+// - `none`: This setting will cause auto-import to never merge use-trees keeping them as simple
+//  paths.
+//
+// In `VS Code` the configuration for this is `rust-analyzer.assist.importMergeBehaviour`.
+//
+// .Import Prefix
+//
+// The style of imports are configurable through the `importPrefix` setting.
```
Comment (lnicola) [inline crates/assists/src/handlers/auto_import.rs:52]:
```suggestion
// The style of imports in the same crate is configurable through the `importPrefix` setting.
```

**Comment by matklad** [comment]
bors d+

**Pull request merged**